doc/internals/zarr-encoding-spec.rst

```diff
@@ -22,23 +22,58 @@ NetCDF and Zarr is that all NetCDF arrays have *dimension names* while Zarr
 
-To accomplish this, Xarray developers decided to define a special Zarr array
-attribute: ``_ARRAY_DIMENSIONS``. The value of this attribute is a list of
-dimension names (strings), for example ``["time", "lon", "lat"]``. When writing
-data to Zarr, Xarray sets this attribute on all variables based on the variable
-dimensions. When reading a Zarr group, Xarray looks for this attribute on all
-arrays, raising an error if it can't be found. The attribute is used to define
-the variable dimension names and then removed from the attributes dictionary
-returned to the user.
-
-Because of these choices, Xarray cannot read arbitrary array data, but only
-Zarr data with valid ``_ARRAY_DIMENSIONS`` or
-`NCZarr <https://docs.unidata.ucar.edu/nug/current/nczarr_head.html>`_ attributes
-on each array (NCZarr dimension names are defined in the ``.zarray`` file).
-
-After decoding the ``_ARRAY_DIMENSIONS`` or NCZarr attribute and assigning the variable
-dimensions, Xarray proceeds to [optionally] decode each variable using its
-standard CF decoding machinery used for NetCDF data (see :py:func:`decode_cf`).
+Dimension Encoding in Zarr Formats
+-----------------------------------
+
+Xarray encodes array dimensions differently depending on the Zarr format version:
+
+**Zarr V2 Format:**
+Xarray uses a special Zarr array attribute: ``_ARRAY_DIMENSIONS``. The value of this
+attribute is a list of dimension names (strings), for example ``["time", "lon", "lat"]``.
+When writing data to Zarr V2, Xarray sets this attribute on all variables based on the
+variable dimensions. This attribute is visible when accessing arrays directly with
+zarr-python.
+
+**Zarr V3 Format:**
+Xarray uses the native ``dimension_names`` field in the array metadata. This is part
+of the official Zarr V3 specification and is not stored as a regular attribute.
+When accessing arrays with zarr-python, this information is available in the array's
+metadata but not in the attributes dictionary.
+
+When reading a Zarr group, Xarray looks for dimension information in the appropriate
+location based on the format version, raising an error if it can't be found. The
+dimension information is used to define the variable dimension names and then
+(for Zarr V2) removed from the attributes dictionary returned to the user.
+
+Coordinate Encoding
+-------------------
+
+In addition to dimension information, Xarray follows CF conventions when encoding
+coordinate metadata. This results in a ``coordinates`` attribute that lists the
+names of coordinate variables associated with each data variable. This attribute
+is separate from and complementary to the dimension encoding described above.
+
+The ``coordinates`` attribute:
+
+- Lists coordinate variables (e.g., ``"yc xc"`` for spatial coordinates)
+- Is visible in the attributes when accessing arrays with zarr-python
+- Is present regardless of Zarr format version
+- Follows CF conventions for coordinate metadata
```

Comment:
I don't think it makes sense to include this section. We should note that Xarray uses it's standard CF encoding/decoding functionality (which you do below), but the `coordinates` attribute is only one part of CF conventions.

Reply:
Yes, a short mention of conventions would definitely be welcome

Reply:
updated